recipes/coal/meta.yaml

```diff
@@ -0,0 +1,123 @@
+{% set name = "coal" %}
+{% set cppname = "lib" + name %}
+{% set pythonname = name + "-python" %}
+{% set version = "3.0.0" %}
+
+package:
+  name: {{ name|lower }}-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/coal-library/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 6a9cbd4684e907fd16577e5227fbace06ac15ca861c0846dfe5bc81e565fb1e7
+
+build:
+  number: 0
+
+outputs:
+  - name: {{ cppname }}
+    build:
+      run_exports:
+        - {{ pin_subpackage(cppname, max_pin='x.x.x') }}
+    script: build_cppcoal.sh   # [not win]
+    script: build_cppcoal.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - {{ stdlib("c") }}
+        - clang  # [win]
+        - cmake
+        - ninja
+        - doxygen
+      host:
+        - eigen
+        - assimp
+        - octomap
+        - libboost-devel
+        - qhull
+        - qhull-static  # [not win]
+      run:
+        - libboost-devel
+        # Since qhullcpp is leaking in header file
+        # consumer will need it to build against coal
+        - qhull-static  # [not win]
+    test:
+      requires:
+        - pkg-config
+      commands:
+        - pkg-config --modversion {{ name }}
+        - test -f $PREFIX/lib/lib{{ name }}${SHLIB_EXT}  # [unix]
+
+  - name: {{ pythonname }}
+    script: build_coal_python.sh   # [not win]
+    script: build_coal_python.bat  # [win]
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - numpy                                  # [build_platform != target_platform]
+        - {{ compiler('cxx') }}
+        - {{ stdlib("c") }}
+        - clang  # [win]
+        - cmake
+        - ninja
+        - doxygen
+        - pylatexenc  # [not win]
+        - lxml  # [not win]
+      host:
+        - eigen
+        - assimp
+        - octomap
+        - libboost-devel
+        - qhull
+        - qhull-static  # [not win]
+        - python
+        - libboost-python-devel
+        - eigenpy
+        # On Windows, Numpy 2.0 don't allow to export the PY_ARRAY_UNIQUE_SYMBOL.
+        # This has been fixed in Numpy 2.1.
+        - numpy >=2.1      # [win and py>=310]
+        # Unfortunatly, Numpy 2.1 is not distributed with Python 3.9.
+        # We are forced to use Numpy 1 in this case.
+        - numpy >=1.19,<2  # [win and py<310]
+        - numpy            # [not win]
+        - {{ pin_subpackage(cppname, exact=true) }}
+      run:
+        - python
+        - numpy >=2.1                    # [win and py>=310]
+        - {{ pin_compatible('numpy') }}  # [win and py<310]
```

Comment:
Please don't use this anymore
```suggestion
```

Reply:
Hello,
as explained in the comment, eigenpy can't build with Numpy 2.0 on Windows. We need a new feature added in Numpy 2.1.
Also, Numpy 2.1 is not distributed for Python 3.9 on Windows, so the only way to provide our package for Python 3.9 on Windows is to build against Numpy 1 and then, use the pin_compatible command.